internals/overlord/stateengine.go

```diff
@@ -72,12 +80,42 @@ func (se *StateEngine) State() *state.State {
 	return se.state
 }
 
-type ensureError struct {
+type multiError struct {
```

Comment:
Related to the other point: this was a semantic error (an error in ensure, specifically), and it's now just a grouping. If we want to generalize this, should we be using the "new" errors package (new for the lifecycle of this system, not as in recent)?  Are there any places catching this error that are actually trying to handle ensure errors, or is it always just grouping?  Let's please check this in snapd as well.

Reply:
This indeed is not a good solution. However, I think we could replace `ensureError`/`dryStartError` with snapd's `multiError`, since the `newMultiError(header, errs string)` function forces you to provide a "primary" error message.

In our case, I haven't seen anywhere in the code where we handle `multiError`s meaningfully, we either discard the function return value or discriminate `multiError`s only to construct a new one.

In snapd, only two functions create new `multiError`s: `CleanupAbandonedImports()` and `(t *importTransaction) Cancel()`. The `multiError` of the former is logged, while in the latter errors are not handled in any case.

Reply:
Okay, so here is a suggested algorithm for the Error method on multiError:

1) If it's a single error, return `<prefix>+<errs[0]>`
2) With more than one, return: `Join([<prefix>, errs...], "\n<two spaces>-<one space>")`

That is, with len(errs) == 1, we get:
```
cannot do foo bar: something bad
```
while len(errs) > 1 become:
```
cannot do foo bar:
  - something bad
  - something else also bad
```